app/actions/websocket/posts.ts

```diff
@@ -136,8 +138,14 @@ export async function handleNewPostEvent(serverUrl: string, msg: WebSocketMessag
             } else if ((post.channel_id === currentChannelId)) { // TODO: THREADS && !viewingGlobalThreads) {
                 // Don't mark as read if we're in global threads screen
                 // the currentChannelId still refers to previously viewed channel
-                markAsViewed = false;
-                markAsRead = true;
+
+                const isChannelScreenMounted = EphemeralStore.getNavigationComponents().includes(Screens.CHANNEL);
```

Comment:
Instead of this, shouldn't we be just more careful about `currentChannelId` be only defined when we are indeed on a channel?

Also, now that I am thinking about it, probably if we are in a thread (not CRT), we will be in a channel, and the channel screen is mounted (one layer below the thread), and still we will mark the post as read. Wonder if we should look a bit deeper into that (maybe on a different PR).

Reply:
There are a few options here:
1. When you leave a channel (back to the home screen) remove the current channel Id -> User may wonder why their channel there were just in is no longer visible (collapsed category)
2. is the Channel screen at the top of the stack -> channel will be mark as unread if looking at the thread screen and won't be marked as read until we switch to it again, what happens if you are in the Gallery and so on.

So I do think this is the right approach.